nominal/core/dataset.py

```diff
@@ -185,6 +187,31 @@ def get_channels(
             else:
                 next_page_token = response.next_page_token
 
+    def to_pandas(self, channel_exact_match: Sequence[str] = (), channel_fuzzy_search_text: str = "") -> pd.DataFrame:
+        """Download a dataset to a pandas dataframe, optionally filtering for only specific channels of the dataset.
+        Args:
+            channel_exact_match: Filter the returned channels to those whose names match all provided strings (case insensitive).
+                For example, a channel named 'engine_turbine_rpm' would match against ['engine', 'turbine', 'rpm'],
+                whereas a channel named 'engine_turbine_flowrate' would not!
+            channel_fuzzy_search_text: Filters the returned channels to those whose names fuzzily match the provided string.
+        Returns:
+            A pandas dataframe whose index is the timestamp of the data, and column names match those of the selected channels.
+
+        Example:
+        ```
+        import nominal as nm
+
+        rid = "..." # Taken from the UI or via the SDK
+        dataset = nm.get_dataset(rid)
+        s = dataset.to_pandas()
+        print("index:", s.index, "index mean:", s.index.mean())
+        ```
+        """
+        rid_name = {ch.rid: ch.name for ch in self.get_channels(channel_exact_match, channel_fuzzy_search_text)}
+        body = _get_series_values_csv(self._clients.auth_header, self._clients.dataexport, rid_name)
+        df = pd.read_csv(body, parse_dates=["timestamp"], index_col="timestamp")
```

Comment:
I wonder what the backend does here if you also upload a dataset with a column named "timestamp"...